File: code/eprogramoflifeapp/app/src/main/java/domain/YearlyCommitment.java
package domain;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * Created by mmfsf on 3/29/17.
 */

public class YearlyCommitment extends Commitment {

    public YearlyCommitment(String name) {
        super(name);
        this.frequency = Frequency.Yearly;
        simpleDateFormat = new SimpleDateFormat("yyyy");
    }

    public Level getPerformed(Date date)
    {
        String key = simpleDateFormat.format(date);

        Level level = this.performed.get(key);
        if(level == null)
        {
            level = Level.NotDone;
        }
        return level;
    }

    public void Point(Date date, Level level)
    {
        String key = simpleDateFormat.format(date);
        this.performed.put(key, level);
    }
}

File: code/eprogramoflifeweb/epl.core/Domain/YearlyCommitment.cs
using epl.core.ValuesObjects;
using System;
using System.Collections.Generic;
using System.Text;

namespace epl.core.Domain
{
  public class YearlyCommitment : Commitment
  {
    protected override string KeyFormat { get => "yyyy"; }

    public YearlyCommitment(string name) : base(name)
    {
      this.Frequency = Frequency.Yearly;
    }

    public YearlyCommitment(string name, string description) : base(name, description)
    {
      this.Frequency = Frequency.Daily;
    }

    public override Level GetPoint(DateTime date)
    {
      var key = date.Date.ToString(this.KeyFormat);
      return this.Performed.ContainsKey(key) ? this.Performed[key] : Level.NotDone;
    }

    public override void Point(DateTime date, Level level)
    {
      var key = date.Date.ToString(this.KeyFormat);
      if (this.Performed.ContainsKey(key))
      {
        this.Performed[key] = level;
      }
      else
      {
        this.Performed.Add(key, level);
      }
    }

  }
}